domain/schema_checker.go

```diff
@@ -44,22 +51,27 @@ func NewSchemaChecker(do *Domain, schemaVer int64, relatedTableIDs []int64) *Sch
 }
 
 // Check checks the validity of the schema version.
-func (s *SchemaChecker) Check(txnTS uint64) error {
+func (s *SchemaChecker) Check(txnTS uint64) (*tikv.RelatedSchemaChange, error) {
+	return s.CheckBySchemaVer(txnTS, intSchemaVer(s.schemaVer))
+}
+
+// CheckBySchemaVer checks if the schema version valid or not at txnTS.
+func (s *SchemaChecker) CheckBySchemaVer(txnTS uint64, startSchemaVer tikv.SchemaVer) (*tikv.RelatedSchemaChange, error) {
```

Comment:
Why we need to return `*tikv.RelatedSchemaChange`? Seems we don't use it.

Reply:
It's a implementation for `schemaLeaseChecker` in `2pc.go`, the returned `tikv.RelatedSchemaChange` will be used in transaction amender in following pull requests.

Reply:
Got it.